util/util.go

```diff
@@ -0,0 +1,53 @@
+package util
+
+import (
+	"strings"
+)
+
+// Logger is used by Session to write logs
+type Logger interface {
```

Comment:
Why not just duplicate this interface into each package...

The idea with golang interfaces is that they are implicitly implemented.